src/server/game/Spells/Auras/SpellAuraEffects.cpp

```diff
@@ -5476,6 +5476,32 @@ void AuraEffect::HandleBindSight(AuraApplication const* aurApp, uint8 mode, bool
     caster->ToPlayer()->SetViewpoint(target, apply);
 }
 
+void AuraEffect::HandleFarSight(AuraApplication const* /*aurApp*/, uint8 mode, bool apply) const
+{
+    if (!(mode & AURA_EFFECT_HANDLE_REAL))
+    {
+        return;
+    }
+
+    Unit* caster = GetCaster();
+    if (!caster || caster->GetTypeId() != TYPEID_PLAYER)
+    {
+        return;
+    }
+
+    Player* player = caster->ToPlayer();
+    if (apply)
```

Comment:
if player can be null I would suggest adding a null check, what do you think?

Reply:
No, it can't be.
```
if (!caster || caster->GetTypeId() != TYPEID_PLAYER)
    {
        return;
    }
 ````